### PR TITLE
teacher dashboard created

### PR DESCRIPTION
**Pull request by rishitashaw**
following tabs are created
1. applications
2. posts
3. create new post
4. profile (ongoing)
5. dashboard (ongoing)

issues fixed
1. eslint config changed so that the error delete 'cr' [prettier/prettier] doesn't arise
2. separate folder for routes, context and library created which can be later used for refactoring code.

### CODE REVIEW

File: src/Components/teacherDashboard/applications.jsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    status: 'accepted',
+  },
+  {
+    id: 9,
+    lastName: 'Roxie',
+    firstName: 'Harvey',
+    age: 65,
+    status: 'accepted',
+  },
+];
```
Comment (SantamRC) [inline src/Components/teacherDashboard/applications.jsx]: Use the Data folder to store temporary data. Make a new json file there and import data from there.

File: src/Components/teacherDashboard/cards.jsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  },
+  expandOpen: {
+    transform: 'rotate(180deg)',
+  },
+  avatar: {
+    backgroundColor: red[500],
+  },
+}));
+
+export default function RecipeReviewCard() {
```
Comment (SantamRC) [inline src/Components/teacherDashboard/cards.jsx]: Use relevant function names

File: src/constants/routes.jsx [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/* eslint-disable import/prefer-default-export */
+export const DASHBOARD_TEACHER = '/teacher/dashboard';
```
Comment (SantamRC) [inline src/constants/routes.jsx]: I don't think a separate route file is needed here. You can just use the path in the react router segment in App.jsx

**Comment by rishitashaw** [comment]
previously mentioned issues fixed
1. useAuthListener custom hook added so that if the user was previously logged in from the device doesn't need to log in again
2. two custom routes added so that if the user isn't logged in he/she can't access the browse page and other one can be used so that the user don't have to signin each time if he/she didn't sign out the last time

suggestions:

1. remove , on line 20 of .eslintrc.json file after "endOfLine": "auto". it's creating unnecessary conflicts and without that the project isn't working
2. we can use useHistory hook for saving browser history


questions

1. do i need to make an edit profile page for professors?
2. what should be present in the main dashboard of profs?

**Comment by SantamRC** [comment]
Make the following changes: -
1. The Dashboard should have two buttons - Profile and Projects
2. The Profile Page will have some informations of the Professor- Profile Picture, Name, Qualification, Affiliation

The Projects Page:-
1.  It will have all the projects that the Professor has uploaded with a Delete and Details button on each of them. 
2. When details button is clicked it will take the professor to another Page where the Project Details will be there and all the students applied will be there. 
3. Each student card will have a Reject button on them.
2. In the Projects page ,at the top there will be a button- Add new Project
3. Clicking on the button will take it to another Page having a form where the Professor can add a new project
4. The form will have the following fields- Project Title, Description, Skills Required, Eligibility

The main page of the dashboard can have updates that the Professor has missed since the last session.

PS: I thing a sidebar is not needed for this layout. You can remove it and keep the Profile and Projects Buttons on Top of the Dashboard Page.

**Pull request merged**